src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,260 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Net.Http
+{
+    public sealed class HttpEnvironmentProxyCredientials : ICredentials
+    {
+        // Wrapper class for cases when http and https has different authentication.
+        private readonly NetworkCredential _http;
+        private readonly NetworkCredential _https;
+
+        public HttpEnvironmentProxyCredientials(NetworkCredential http, NetworkCredential https)
+        {
+            _http = http;
+            _https = https;
+        }
+
+        public NetworkCredential GetCredential(Uri uri, string authType)
+        {
+            return uri.Scheme == Uri.UriSchemeHttp ? _http : _https;
+        }
+    }
+
+    internal sealed class HttpEnvironmentProxy : IWebProxy
+    {
+        private const string EnvAllProxyUC = "ALL_PROXY";
+        private const string EnvAllProxyLC = "all_proxy";
+        private const string EnvHttpProxyLC = "http_proxy";
+        private const string EnvHttpsProxyLC = "https_proxy";
+        private const string EnvHttpsProxyUC = "HTTPS_PROXY";
+        private const string EnvNoProxyLC = "no_proxy";
+
+        private Uri _http;          // String URI for HTTP requests
```

Comment:
Can we call these something like _httpProxy and _httpsProxy?  Just "_http" isn't super descriptive.